hack/golangci-kal/.custom-gcl.yaml

```diff
@@ -0,0 +1,6 @@
+version: v2.4.0
```

Comment:
It looks like we can skip this file, please check my comment: https://github.com/kubeflow/trainer/pull/2858#discussion_r2382247375

Reply:
I think I still need this. Someone needs to build the golangci-lint-kai linter.

We can either use golangci plugins and build the so from the repo.

But that is considered deprecated from golangci or we follow the approach where we build a binary and run a custom integration.